rpc-tests/tests/test-rpc.js

```diff
@@ -21,69 +25,80 @@ describeWithNetwork(network, `${network} RPC`, function(context) {
-        const api = context.api;
-
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-      
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const alice = keyring.addFromUri('//Alice');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.dappsStaking.register(getAddressEnum(CONTRACT));
-      
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(alice, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
-
-        expect(hash).to.exist;
+	it('should be able to Register contract on H160 address 0x01 using Alice account', async () => {
+        const events = await sendTransaction(
+            context.api,
+            context.api.tx.dappsStaking.register(getAddressEnum(CONTRACT)),
+            context.alice,
+            'dappsStaking',
+            'NewContract'
+        );
+
+        expect(events.length).to.equal(1);
+        expect(events[0].event.section).to.equal('dappsStaking');
+        expect(events[0].event.method).to.equal('NewContract');
+        expect(events[0].event.data[0].toString()).to.equal(ALICE);
     });
 
-	it('should be able to transfer tokens from alice to bob', async function () {
-        const api = context.api;
-
-        // Construct the keyring after the API (crypto has an async init)
-        const keyring = new Keyring({ type: 'sr25519' });
-      
-        // Add Alice to our keyring with a hard-derivation path (empty phrase, so uses dev)
-        const alice = keyring.addFromUri('//Alice');
-
-        // Create a extrinsic, transferring 100 units to Bob
-        const transfer = api.tx.balances.transfer(BOB, 100);
-      
-        // Sign and send the transaction using our account
-        const hash = await transfer.signAndSend(alice, { nonce: -1 });
-
-        console.log('Transfer sent with hash', hash.toHex());
-
-        expect(hash).to.exist;
+	it('should be able to transfer tokens from alice to bob', async () => {
+        const events = await sendTransaction(
+            context.api,
+            context.api.tx.balances.transfer(BOB, 100),
+            context.alice,
+            'balances',
+            'Transfer'
+        );
+
+
+        const filtered = events.filter((eventObj) => {
+            const { event: { method, section, data } } = eventObj;
+
+            return (
+                section === 'balances' &&
+                method === 'Transfer' &&
+                data[0].toString() === ALICE &&
+                data[1].toString() === BOB &&
+                data[2].toString() === '100'
+            );
+        });
+
+        // Check filtered event
+        expect(filtered.length).to.equal(1);
+        expect(filtered[0].event.section).to.equal('balances');
+        expect(filtered[0].event.method).to.equal('Transfer');
```

Comment:
Same comment as for dapps-staking register - imo, it would be better to check states before & after executing transactions.

What is your opinion on this?

Reply:
I agree, we should be checking states. Adding state check.

Reply:
Done